packages/piro/test/MockModelEval_A_Tpetra.cpp

```diff
@@ -320,13 +320,18 @@ void MockModelEval_A_Tpetra::evalModelImpl(
   auto p = p_vec->getData();
 
   if (f_out != Teuchos::null) {
+    f_out->putScalar(0.0);
     for (int i=0; i<myVecLength; i++) {
       int gid = x_in->getMap()->getGlobalElement(i);
 
-      if (gid==0) // f_0 = (x_0)^2 - p_0
-       f_out->getDataNonConst()[i] = x[i] * x[i] -  p[0];
-      else // f_i = x_i^2 - (i+p_1)^2 (for i != 0)
-       f_out->getDataNonConst()[i] = x[i] * x[i] - (gid + p[1])*(gid + p[1]);
+      if (gid==0) {// f_0 = (x_0)^3 - p_0
+       f_out->getDataNonConst()[i] = x[i] * x[i] * x[i] -  p[0];
```

Comment:
Please lift the `getDataNonConst()` calls outside the loop.  Those are expensive, and will get VERY expensive once we get rid of the UVM assumption.

Reply:
Thanks. These are tiny vectors in a test. Anyway I'll fix that.